src/main/java/seedu/address/logic/commands/AddInsuranceCommand.java

```diff
@@ -38,8 +47,29 @@ public AddInsuranceCommand(Index index, int insuranceID) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        throw new CommandException(
-                String.format(MESSAGE_ARGUMENTS, index.getOneBased(), insuranceID));
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        try {
+            InsurancePlansManager personToEditInsurancePlansManager = personToEdit.getInsurancePlansManager();
+
+            InsurancePlan planToBeAdded = InsurancePlanFactory.createInsurancePlan(insuranceID);
+
+            personToEditInsurancePlansManager.checkIfPlanNotOwned(planToBeAdded);
+
+            personToEditInsurancePlansManager.addPlan(planToBeAdded);
+
+            return new CommandResult(String.format(MESSAGE_ADD_INSURANCE_PLAN_SUCCESS,
+                    personToEditInsurancePlansManager, Messages.format(personToEdit)));
```

Comment:
Formatting incorrect, for example it shows:
Added Insurance Plan: Insurance Plans: Travel Insurance Plan, to Client: Alex Yeoh; Phone: 12345678; Email: alexyeoh@example.com; Address: Blk 30 Geylang Street 29, #06-40; Tags: [friends]

Will edit this again later, just commenting for reference